app/controllers/items_controller.rb

```diff
@@ -16,4 +17,25 @@ def create
       flash[:error] = "There was an error saving the item.  Please try again."
     end
   end
+
+  def destroy
+    @user = current_user
+    @item = Item.find(params[:id])
+    @item.destroy
+    flash[:notice] = "Item was removed from your list - Congratulations!"
+    redirect_to user_path(@user)
+  end
+
+  private
+
+  def item_params
+    params.require(:item).permit(:description)
+  end
+
+  def only_permit_current_user
+    unless user_signed_in? && (@user == current_user)
```

Comment:
3 steps for an AJAX call:

1. the view needs to know to make a remote AJAX call when a link is clicked
or a form is submitted, which means it needs `remote: true`

2. the action in the controller needs to know that it should be able to
respond to an AJAX request, which means it needs some sort of `respond_to`
block with an option for `js`

3. you need a js view template with what js that will be executed once the
request is successful, which means you need a file titled `action.js.erb`
where "action" is the name of the controller action (e.g. "create") and, in
that file, you need your js to change the view in some way (e.g. by
appending to something, or prepending to something, or removing something,
or adding something, etc...)

On Tue, Jan 12, 2016 at 8:25 AM, Aaron Kozminski <notifications@github.com>
wrote:

> In app/controllers/items_controller.rb
> <https://github.com/aaronkoz74/blocitoff/pull/8#discussion_r49459225>:
>
> > +  def destroy
> > +    @user = current_user
> > +    @item = Item.find(params[:id])
> > +    @item.destroy
> > +    flash[:notice] = "Item was removed from your list - Congratulations!"
> > +    redirect_to user_path(@user)
> > +  end
> > +
> > +  private
> > +
> > +  def item_params
> > +    params.require(:item).permit(:description)
> > +  end
> > +
> > +  def only_permit_current_user
> > +    unless user_signed_in? && (@user == current_user)
>
> Got it to work. For some reason I can't get the ajax function to work
> properly ... removed the `redirect_to` portion in the `destroy` method and
> my page isn't refreshing without a manual command. Talk to you in a while.
> Thanks
> … <#1991570044_>
> On Mon, Jan 11, 2016 at 11:27 PM, bjblock <notifications@github.com>
> wrote: In app/controllers/items_controller.rb <#8 (comment)
> <https://github.com/aaronkoz74/blocitoff/pull/8#discussion_r49414918>>: >
> + def destroy > + @user <https://github.com/user> = current_user > + @item
> <https://github.com/item> = Item.find(params[:id]) > + @item.destroy > +
> flash[:notice] = "Item was removed from your list - Congratulations!" > +
> redirect_to user_path(@user <https://github.com/user>) > + end > + > +
> private > + > + def item_params > +
> params.require(:item).permit(:description) > + end > + > + def
> only_permit_current_user > + unless user_signed_in? && (@user
> <https://github.com/user> == current_user) we can go over this tomorrow.
> — Reply to this email directly or view it on GitHub <
> https://github.com/aaronkoz74/blocitoff/pull/8/files#r49414918>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aaronkoz74/blocitoff/pull/8/files#r49459225>.
>